core/src/actors/chain_manager/handlers.rs

```diff
@@ -45,9 +45,33 @@ impl Handler<EpochNotification<EpochPayload>> for ChainManager {
 impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
     type Result = ();
 
-    fn handle(&mut self, msg: EpochNotification<EveryEpochPayload>, _ctx: &mut Context<Self>) {
+    fn handle(&mut self, msg: EpochNotification<EveryEpochPayload>, ctx: &mut Context<Self>) {
         debug!("Periodic epoch notification received {:?}", msg.checkpoint);
         self.current_epoch = Some(msg.checkpoint);
+
+        if let Some(candidate) = self.block_candidate.clone() {
```

Comment:
You can use the `take()` method to return the `Block` and replace the value of an `Option` with `None`.

```suggestion
        if let Some(candidate) = self.block_candidate.take() {
```